core/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -178,10 +190,22 @@ private[chisel3] trait HasId extends InstanceId {
   }
 
   /** This resolves the precedence of [[autoSeed]] and [[suggestName]]
+    *
+    * @note It will become an error in the future to suggestName the same thing that autoSeed would have assigned
     *
     * @return the current calculation of a name, if it exists
     */
-  private[chisel3] def seedOpt: Option[String] = suggested_seed.orElse(auto_seed)
+  private[chisel3] def seedOpt: Option[String] = {
+    suggested_seed.zip(auto_seed).foreach {
+      case (suggested, auto) =>
+        if (suggested == auto) {
+          Builder.deprecated(
+            s"calling suggestName(${suggested}) had no effect as it is the same as the auto prefixed name, this will become an error in 3.6"
```

Comment:
```suggestion
            s"calling .suggestName(\"${suggested}\") had no effect as it is the same as the automatically given name, this will become an error in 3.6"
```

So this warning has a bit of a problem. For test `3` in your tests, the message you get is:
```
[deprecated] DependencyManager.scala:280 (1 calls): calling suggestName(wire) had no effect as it is the same as the auto prefixed name, this will become an error in 3.6
```

We can't give a good source locator here so we should make sure to give no source locator at all. Also we need to give the `.toString` for the `Data` so that it will at least let the user know what module it is in. Not having a source locator is pretty bad though....